### PR TITLE
AG-108 Lepsie zabezpecenie GitHub Secrets pri packaging-u

### PR DESCRIPTION
**Pull request by pomali**
- na package action je samostatny environment `packaging` do ktoreho su presunute secretes
- na zmenu workflows a package.sh je potrebny suhlas od `@slovensko-digital/autogram-packaging` team-u
- tym padom by nemali byt pristupne secrets v testing workflowoch

TODO:
- vytvorit environment `packaging` a presunut tam secrets
- vytvorit `@slovensko-digital/autogram-packaging` team

### CODE REVIEW

**Comment by jsuchal** [comment]
Spravil som @slovensko-digital/autogram-release-team a `packaging` env. secrets zatial este nepresunute.

File: .github/workflows/package.yaml [hunk 1]
```diff
@@ -8,6 +8,7 @@ on:
 jobs:
   build:
     runs-on: ${{ matrix.config.os }}
+    environment: packaging
```
Comment (jsuchal) [inline .github/workflows/package.yaml:11]: Spravanie tu je ake? "Dedi" to ine envars alebo pouziva len envarss z toho environemntu?

**Review by jsuchal — COMMENTED** [review]
Tu este pridajme, ze nech sa pustaju testy aj pre PR ked toto teda zafunguje.

File: .github/workflows/package.yaml (hunk 1 above)
Comment (pomali) [reply .github/workflows/package.yaml:11]: dedi sa to s tym, najnizsi level overriduje

> If a secret with the same name exists at multiple levels, the secret at the lowest level takes precedence. For example, if an organization-level secret has the same name as a repository-level secret, then the repository-level secret takes precedence. Similarly, if an organization, repository, and environment all have a secret with the same name, the environment-level secret takes precedence.

**Comment by pomali** [comment]
Pridal som sem aj Windows signing
pokusny run v https://github.com/solver-it-sro/autogram/actions/runs/5255461697

File: .github/workflows/package.yaml [hunk 2]
```diff
@@ -99,15 +113,36 @@ jobs:
...
-          publish_only_summary: false
+          name: target
+          path: |
+            target/*.msi
+
+  sign_windows:
+    name: Sign Windows installer (Windows)
+    needs: build
+    runs-on: windows-latest
+    environment: sign-windows
```
Comment (jsuchal) [inline .github/workflows/package.yaml:126]:
```suggestion
    environment: packaging-windows
```

File: .github/workflows/test.yaml [hunk 3]
```diff
@@ -1,14 +1,16 @@
 name: Tests
 
-on: [push]
+on:
+  push:
+  pull_request:
+    branches:
+      - main
```
Comment (jsuchal) [inline .github/workflows/test.yaml:7]:
```suggestion
  pull_request:
```

File: .github/workflows/package.yaml (hunk 1 above)
Comment (jsuchal) [inline .github/workflows/package.yaml:11]:
```suggestion
    environment: packaging
    
    permissions:
      checks: write
```

File: .github/workflows/package.yaml [hunk 4]
```diff
@@ -71,7 +72,20 @@ jobs:
...
+          coverage_results_path: ${{ github.workspace }}/target/site/jacoco/jacoco.xml
+          coverage_report_name: Coverage
+          coverage_report_title: JaCoCo
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          skip_check_run: false
+          minimum_coverage: 80
+          fail_below_threshold: false # set to true for Github CI build failing when coverage check do not pass
+          publish_only_summary: false
+
+      - name: Notarize release with Apple (MacOS)
```
Comment (jsuchal) [inline .github/workflows/package.yaml:88]:
```suggestion
      - name: Notarize release with Apple (MacOS)
```

**Pull request merged**